nvflare/fuel/hci/cmd_arg_utils.py

```diff
@@ -68,3 +71,82 @@ def get_usage(self) -> str:
         usage_output = buffer.getvalue().split("\n", 1)[1]
         buffer.close()
         return usage_output
+
+
+def process_targets_into_str(targets: List[str]) -> str:
+    if not isinstance(targets, list):
+        raise SyntaxError("targets is not a list.")
+    if not all(isinstance(t, str) for t in targets):
+        raise SyntaxError("all targets in the list of targets must be strings.")
+    for t in targets:
+        try:
+            validate_required_target_string(t)
+        except SyntaxError:
+            raise SyntaxError("each target in targets must be a string of only valid characters and no spaces.")
```

Comment:
suggest include what target failed and include the error message raised from validate_required_target_string:

```
except SyntaxError as e:
    raise SyntaxError(f"target: {t} is invalid: {e}")
```

Reply:
right

Reply:
done as suggested

Reply:
The current code:

```
except SyntaxError:
            raise SyntaxError("each target in targets must be a string of only valid characters and no spaces.")
```

Users can NOT know which target is having issue, as we swallow individual SyntaxError and just raise this overall one.

Reply:
This has been fixed already.